ipn/ipnlocal/local.go

```diff
@@ -4048,11 +3980,57 @@ func (b *LocalBackend) SwitchProfile(profile ipn.ProfileID) error {
 	return b.resetForProfileChangeLockedOnEntry()
 }
 
+func (b *LocalBackend) initTKALocked() error {
+	cp := b.pm.CurrentProfile()
+	if cp.ID == "" {
+		b.tka = nil
+		return nil
+	}
+	if b.tka != nil {
+		if b.tka.profile == cp.ID {
+			// Already initialized.
+			return nil
+		}
+		b.tka = nil // reset
```

Comment:
Its not clear to me what this does: Is this meant to clear a TKA state from a previous profile once you switch?

Reply:
yep, added a comment.